server/routes/incidentDetails/incidentDetails.ts

```diff
@@ -42,40 +72,101 @@ export default class IncidentDetailsRoutes {
     const locations = await this.locationService.getIncidentLocations(agencyId, user)
 
     const data = {
-      incidentDate: this.getIncidentDate(incidentDate),
-      locationId,
+      incidentDate: this.getIncidentDate(incidentDate) || this.getIncidentDate(req.session.incidentDate),
+      locationId: locationId || req.session.incidentLocation,
+    }
+
+    const associatedPersonDetails = { id: selectedPerson, name: '' }
+    if (selectedPerson) {
+      const associatedPrisoner = await this.placeOnReportService.getPrisonerDetails(selectedPerson, user)
+      associatedPersonDetails.name = associatedPrisoner.displayName
     }
 
     return res.render(`pages/incidentDetails`, {
       errors: error ? [error] : [],
+      prisonerNumber,
       prisoner,
       locations,
       data,
       reportingOfficer: reporter || '',
       submitButtonText: 'Save and continue',
       reportPreviouslySubmitted: false,
+      queryRadioSelection,
+      inciteAnotherPrisonerInput,
+      assistAnotherPrisonerInput,
+      associatedPersonDetails,
     })
   }
 
   view = async (req: Request, res: Response): Promise<void> => this.renderView(req, res, {})
 
   submit = async (req: Request, res: Response): Promise<void> => {
-    const { incidentDate, locationId } = req.body
+    const {
+      incidentDate,
+      locationId,
+      currentRadioSelected,
+      search,
+      inciteAnotherPrisonerInput,
+      assistAnotherPrisonerInput,
+    } = req.body
     const { user } = res.locals
     const { prisonerNumber } = req.params
+    const selectedPerson = JSON.stringify(req.query.selectedPerson)?.replace(/"/g, '')
+    const queryRadioSelection = JSON.stringify(req.query.queryRadioSelection)?.replace(/"/g, '')
 
-    const error = validateForm({ incidentDate, locationId })
+    if (search) {
+      const searchValue =
+        search === 'inciteAnotherPrisonerSearchSubmit' ? inciteAnotherPrisonerInput : assistAnotherPrisonerInput
+      const error = validatePrisonerSearch({ searchTerm: searchValue, inputId: currentRadioSelected })
+      if (error)
+        return this.renderView(req, res, {
+          error,
+          incidentDate,
+          locationId,
+          queryRadioSelection: currentRadioSelected,
+          inciteAnotherPrisonerInput,
+          assistAnotherPrisonerInput,
+        })
+      req.session.redirectUrl = `/incident-details/${prisonerNumber}?queryRadioSelection=${currentRadioSelected}`
+      req.session.incidentDate = incidentDate
+      req.session.incidentLocation = locationId
+      const searchPageHref = `/select-associated-prisoner/${prisonerNumber}?searchTerm=${searchValue}`
+      return res.redirect(`${searchPageHref}&startUrl=incident-details`)
+    }
+
+    // Check the selectedPerson has not been tampered with in URL
```

Comment:
Interested to know what is going on here - is this in case the user retypes when back on the radio button page?